### PR TITLE
VACMS-16999 VAMC Detail page web components

### PR DESCRIPTION
**Pull request by randimays**
## Summary
Use web components on VAMC detail pages where possible.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16999

## Testing done
Tested locally at these routes:
- /boston-health-care/make-an-appointment/
- /boston-health-care/contact-us/
- /boston-health-care/about-us/
- /boston-health-care/work-with-us/
- /minneapolis-health-care/make-an-appointment/
- /minneapolis-health-care/contact-us/
- /minneapolis-health-care/about-us/
- /minneapolis-health-care/work-with-us/

Also tested a page [**via Tugboat**](https://web-v9l0sy4euftr8m5pebq4pxezfbvbxcnz.demo.cms.va.gov/western-colorado-health-care/about-us/important-useful-information/) for the process list web component.

## Screenshots

<details><summary>Additional info inside alert component</summary>

### Desktop
Screenshot 2024-05-22 at 4 53 03 PM
Screenshot 2024-05-22 at 4 53 08 PM

### Mobile
Screenshot 2024-05-22 at 4 53 37 PM

### DOM
Screenshot 2024-05-22 at 4 53 30 PM

</details>

<details><summary>Back to top component</summary>
Screenshot 2024-05-22 at 4 53 45 PM
Screenshot 2024-05-22 at 4 53 37 PM

</details>

<details><summary>Downloadable files (on Minneapolis About Us page)</summary>
Screenshot 2024-05-22 at 5 09 57 PM
Screenshot 2024-05-22 at 5 10 30 PM

</details>

<details><summary>Process list</summary>

### Desktop
Screenshot 2024-05-28 at 10 46 28 AM

### Mobile
Screenshot 2024-05-28 at 10 46 33 AM

### DOM
Screenshot 2024-05-28 at 10 46 24 AM

</details>

### CODE REVIEW

**Review by eselkin — COMMENTED** [review]
I remember something about @laflannery asking if when we used the download link the `filetype` should be specified. Is it not possible because we don't know it from drupal. Does it infer it from extension?

**Comment by randimays** [comment]
@eselkin Good callout; I added it here!

**Comment by randimays** [comment]
@eselkin The review instance is back up and ready for your review. CC @laflannery

**Review by eselkin — CHANGES_REQUESTED** [review]
minor tweak

File: src/site/paragraphs/downloadable_file.drupal.liquid [hunk 1]
```diff
@@ -24,36 +24,35 @@
...
-
-        <a class="file-download-with-icon" target="_blank"
-           href="{{url}}" download="{{url}}">
-            <i class="fas fa-download vads-u-margin--0p5" aria-label="Download" role="img"></i>
-            {{entity.fieldTitle}} ({{url | fileExt | upcase}})
-        </a>
+        <va-link
+            download="{{url}}"
+            href="{{url}}"
+            filetype="{{ fileExt }}"
```
[the remark this answers is not in the document]
Comment (eselkin) [reply src/site/paragraphs/downloadable_file.drupal.liquid:30]: Also, if you add the filetype="..."
you can remove the part that uses that in the text="...({{ url|fileExt|upcase }}"

**Comment by randimays** [comment]
@eselkin Laura and I had a discussion about `filetype` in DMs last week. I thought that situation was different from this one but it seems to be the same. When using a `va-link` with `download` and `filetype`, the `filetype` is not read aloud on a screen reader. There is an issue open for this: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2817 @laflannery I see that this issue refers specifically to `channel` and `video` types... should we add `download`?

In short: we can't use the `filetype` because it won't read "(PDF)" until they fix this, so we just have to keep the extension in the `va-link` `text` attribute.

**Comment by randimays** [comment]
@eselkin @laflannery The review instance here is ready to review when you have a moment, please

**Review by eselkin — APPROVED** [review]

**Comment by laflannery** [comment]
@randimays 2 comments:
1. I updated the DST ticket to include download, I meant to but thanks for the reminder!
2. The review instance wasn't working so I used the screenshots and the tugboat. The only difference I saw is that for the BTT, your screenshots look good but the tugboat it's way over to the right, so I'm not sure if that's just an odd Tugboat thing or if something accidently got weird:
Screenshot 2024-05-29 at 8 25 47 AM

File: src/site/paragraphs/downloadable_file.drupal.liquid (hunk 1 above)
[the remark this answers is not in the document]
Comment (randimays) [reply src/site/paragraphs/downloadable_file.drupal.liquid]: I added a comment on the PR about this but realized I should've put it in this thread:

> @eselkin Laura and I had a discussion about filetype in DMs last week. I thought that situation was different from this one but it seems to be the same. When using a va-link with download and filetype, the filetype is not read aloud on a screen reader. There is an issue open for this: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2817
> 
> In short: we can't use the filetype because it won't read "(PDF)" until they fix this, so we just have to keep the extension in the va-link text attribute.

**Comment by randimays** [comment]
@laflannery Back to top component is fixed. Review instance is ready for your review please!

**Review by laflannery — APPROVED** [review]
LGTM!

**Review by ataker — APPROVED** [review]
